src/blop/agent.py

```diff
@@ -402,37 +411,44 @@ def _construct_model(self, obj, skew_dims: list[tuple[int, ...]] | None = None)
                 input_transform=self.input_normalization,
             )
 
-            obj.validity_constraint = GenericDeterministicModel(
-                f=lambda x: obj.validity_conjugate_model.probabilities(x)[..., -1]
-            )
-
     def update_models(
         self,
-        train: bool | None = None,
+        force_train: bool = False,
     ) -> None:
+        """
+        We don't want to retrain the models on every call of everything, but if they are out of sync with
+        the DOFs then we should.
+        """
```

Comment:
Make it clear the behavior of this parameter.
```suggestion
        """
        We don't want to retrain the models on every call of everything, but if they are out of sync with
        the DOFs then we should.
        
        Parameters
        ---------------
        force_train : bool
            Force re-training the model. If false, the model will skip re-training unless there are `self.train_every` more data points to train with.
        """
```